.github/workflows/listener.yml

```diff
@@ -0,0 +1,101 @@
+# Ingest repository_dispatch events and trigger appropriate workflows 
+name: Listener
+
+on:
+  repository_dispatch:
+    types: [Open, Update, Merge, Close]
+
+jobs: 
+  open: 
+    if: ${{ github.event.action == 'Open' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Echo Event
+        run: | 
+          echo "${{ toJson(github) }}"
+      
+      - uses: ./.github/actions/replace_specs
+        id: replace-specs
+        with: 
+          branch-name: ${{ github.event.client_payload.branchName }}
+          username: ${{ secrets.DX_GITHUB_USERNAME }}
+          token: ${{ secrets.DX_GITHUB_TOKEN }}
+      
+      - uses: ./.github/actions/add_pr_comment
+        with: 
+          username: ${{ secrets.DX_GITHUB_USERNAME }}
+          token: ${{ secrets.DX_GITHUB_TOKEN }}
+          repo-name: api-specs
+          pr-number: ${{ github.event.client_payload.prNumber }}
+          message: 'https://bw-api-docs-${{ github.event.client_payload.branchName }}.s3-website-us-east-1.amazonaws.com/\n\nPlease note that it may take a couple minutes for your preview site to become available. See the corresponding PR opened on the internal docsite repository here:\nhttps://github.com/Bandwidth/api-docs/pull/${{ steps.replace-specs.outputs.pr-number }}'
```

Comment:
```suggestion
          message: 'https://bw-api-docs-${{ github.event.client_payload.branchName }}.s3-website-us-east-1.amazonaws.com/\n\nPlease note that it may take a couple minutes for your preview site to become available. See the corresponding PR opened on the docsite repository here:\nhttps://github.com/Bandwidth/api-docs/pull/${{ steps.replace-specs.outputs.pr-number }}'
```